### PR TITLE
fix: Hero image should cover the space not shrink to fit

### PR DESCRIPTION
**Pull request by gabalafou**
This could potentially be a fix for #213, at least a partial fix or a stopgap fix.

Previously, the code was supposedly defaulting to `object-fit: contain` but `cover` matches the [Figma designs](https://www.figma.com/file/vyWM9qXdXCSAPlTfUjuqrL/Quansight-2021-Website?node-id=1181%3A340) much better.

Here are two screenshots showing the difference. Both are at mobile device width.

When object-fit is set to `contain`, the hero image is shrunk down so that it fits, which is not desired:

When object-fit is set to `cover`, the hero image expands across the available space, as desired:

I also had to change the code to not use a default function parameter value for `objectFit` because while debugging, I discovered that the CMS was passing an empty string for `objectFit` when a value for object fit is not chosen in the Storyblok UI.

Here's a snippet of the json returned from `http://localhost:4200/_next/data/development/library.json` in which `objectFit` is assigned an empty string:

```json
{
    "pageProps": {
        "data": {
            "__typename": "PageItem",
            "alternates": [],
            "content": {
                "__typename": "PageComponent",
                "_editable": "<!--#storyblok#{\"name\": \"page\", \"space\": \"147759\", \"uid\": \"52531dbb-5aff-4f6f-a084-fcc2da90c1c1\", \"id\": \"143249754\"}-->",
                "_uid": "52531dbb-5aff-4f6f-a084-fcc2da90c1c1",
                "body": [
                    {
                        "_uid": "5432d60a-c2c8-41a9-9131-13d29e123d97",
                        "image": {
                            "id": 4981932,
                            "alt": "Hero decoration",
                            "name": "",
                            "focus": null,
                            "title": "",
                            "filename": "https://a.storyblok.com/f/147759/1440x746/5cfdc24cd0/paris_v6.png",
                            "copyright": "",
                            "fieldtype": "asset"
                        },
                        "title": "Blog",
                        "variant": "medium",
                        "subTitle": "",
                        "component": "hero",
                        "objectFit": "",
                        "_editable": "<!--#storyblok#{\"name\": \"hero\", \"space\": \"147759\", \"uid\": \"5432d60a-c2c8-41a9-9131-13d29e123d97\", \"id\": \"143249754\"}-->"
                    },
```

And here's a screenshot showing the corresponding empty `Object Fit` field in Storyblok:

### CODE REVIEW

**Comment by bskinn** [comment]
@irinafumarel Is this a valid fix from a design perspective?

If I understood you correctly when we discussed these on Meet last week, I don't think this is consistent with Pierre's vision for the heroes, as data visualizations.

**Comment by gabalafou** [comment]
@bskinn - to be clear, Pierre's vision for the heroes, if you're referring to animated background images, should be an entirely separate ticket from #213, which as I understand it, is only about the ratio and sizing of **static** hero images.

**Comment by bskinn** [comment]
No, agreed, we've abandoned animated hero images.

I believe Pierre does not want the data-containing portions of any of the hero images to be cropped away on any screen size. Irina can confirm/clarify.

**Comment by gabalafou** [comment]
> I believe Pierre does not want the data-containing portions of any of the hero images to be cropped away.

Ah I see. Okay, well just to be clear, that's not at all what was communicated on the Figma boards. For example, here is a screenshot of one of the Figma designs where you can see that the dataviz is cropped at mobile:



I think there have maybe been two competing visions for these data visualizations. 

For Troy, I think they were like art or decoration, in which case you wouldn't worry much about cropping the image. Let's be clear, these are data visualizations, so there is no way to crop them without removing data-containing portions of the image; the entire image contains data. But when it serves as decorative art, it should be okay to crop, since you're less concerned about the data story behind the image and more about the aesthetic value it provides.

For Pierre and others, I think the vision was that the data visualization would draw some attention to itself (through some kind of animated transition) and additionally present some kind of link to a page or a pop-up that explains the visualization and the story that it tells.

I am a huge fan of the second vision, but there has been little concrete work done for it beyond the visualizations themselves, so in my view, this more expansive vision requires a separate project of its own, with allocation of time and resources that go beyond our initial scope of work.

**Comment by irinafumarel** [comment]
@gabalafou @bskinn The data viz was cropped for mobile to illustrate the sizes of the mobile hero so we can upload the mobile hero to Storyblock.  At that time I was communicated that Naturaily will be customizing and animating the heroes so me and Pierre stopped working on this. I will work this week with Pierre to resize that dataviz for mobile.

**Comment by bskinn** [comment]
@gabalafou Agree with everything you wrote here. 

For launch, I think we have to find a solution using static hero images that provides a minimum-attractive solution on all screens (good resolution, no aspect ratio distortion, and good visual layout (no excess padding, etc.)). This may mean we have little choice but to crop the heroes for tablet/mobile, and then figure out a better solution post-launch that avoids cropping.

To be clear, it's entirely possible that existing functionality is sufficient to do this, and we just haven't figured out the right combination of options on the Hero component to achieve it.

To summarize, I think we're looking at a three-stage progression:

1. For launch: A minimum viable product (MVP) Hero solution with static images.
2. Post-launch: If this MVP requires cropping Hero images, developing a solution that does not require cropping of the static images.
3. Post-launch: Developing toward the full "second vision", with animation/transitions and pop-ups leading to explainer pages about each visualization. (One uncertainty here: is this vision compatible with a11y constraints?)

(This might be complex enough to warrant its own Github project/board?)

Comments please!

Crossref #177 and #213.

**Comment by gabalafou** [comment]
No comments to add, just 👍

**Review by marcin-antczak-naturaily — APPROVED** [review]
